src/main/java/com/raisetech09/oka/controller/ProductController.java

```diff
@@ -1,26 +1,39 @@
 package com.raisetech09.oka.controller;
 
-import com.raisetech09.oka.service.Product;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.GetMapping;
-import com.raisetech09.oka.repository.ProductMapper;
-import org.springframework.web.bind.annotation.RestController;
+import com.raisetech09.oka.service.ProductService;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
 @CrossOrigin(origins = "*")
 public class ProductController {
-    private ProductMapper productMapper;
+    private ProductService productService;
 
-    @Autowired
-    public ProductController(ProductMapper productMapper) {
-        this.productMapper = productMapper;
+    public ProductController(ProductService productService) {
+        this.productService = productService;
     }
 
     @GetMapping("/products")
-    public List<Product> get() {
-        return productMapper.findAll();
+    public List<ProductResponse> get() {
+        return productService.findAll();
+    }
+
+    //  idで検索する
+    @GetMapping("/products/{id}")
+    public List<ProductResponse> getProductById(
+            @PathVariable int id
+    ) {
+        List<ProductResponse> product = productService.findById(id);
+        return product;
+    }
+
+    //    localhost:8080/products?price=200のGETリクエストを行い、価格が200で登録されている情報を取得する
+    @GetMapping("/price-list")
```

Comment:
コメントに従うなら
```suggestion
    //    localhost:8080/products?price=200のGETリクエストを行い、価格が200で登録されている情報を取得する
    @GetMapping("/products")
```
じゃないでしょうか？

Reply:
製品を値段で検索するというAPIなので`/products`がよいですね。

Reply:
ご確認ありがとうございます！

`@GetMapping("/products")`が2つある状態だとエラーになったので、統合しました。  
またint型はnullが値が空でもnullが入ってこないことがわかったので、`defaultValue = "-1"`と設定をして、条件分岐によってreturnする内容を変更しました。  

あと、見えている課題として、DBに存在しない値(200000)を入れると`[]`が返却されるので、理想としては、`/products`にリダイレクトさせたほうが使いやすいのかなと思っておりますが、まだ実装方法を考えていない状態です。
おそらく`ProductService.java`でリダイレクトの設定をするのかなぁと考えております。

https://github.com/takahiro-okada/java-raisetech-09/blob/12bb553ffeb1316fb5997e64ecb776e034bf7a87/src/main/java/com/raisetech09/oka/controller/ProductController.java#L17-L28

Reply:
RESTful APIとしては
> 見えている課題として、DBに存在しない値(200000)を入れると[]が返却されるので

これは正しい動きに思えます。
あくまで検索して結果が得られなかったということなので挙動として違和感はないです（この辺画面があるWebアプリケーションと考え方がちょっと違います）。

一方、たとえば、
`/products/1`で指定して対応するリソースがない場合は404エラーとして扱うのが適切だと考えます。

このあたりは調べがいがありますね。
それから、GitHub APIの各APIの仕様書をよんでどのようなエラーハンドリングをしているか参考にするのもいいと思います。